agent/ebs/watcher.go

```diff
@@ -106,12 +124,34 @@ func (w *EBSWatcher) HandleEBSResourceAttachment(ebs *apiebs.ResourceAttachment)
 	volumeId := ebs.GetAttachmentProperties(apiebs.VolumeIdKey)
 	ebsAttachment, ok := w.agentState.GetEBSByVolumeId(volumeId)
 	if ok {
-		log.Infof("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
+		log.Debugf("EBS Volume attachment already exists. Skip handling EBS attachment %v.", ebs.EBSToString())
 		return ebsAttachment.StartTimer(func() {
 			w.handleEBSAckTimeout(volumeId)
 		})
 	}
 
+	// start EBS CSI Driver Managed Daemon
+	if runningCsiTask := w.taskEngine.GetDaemonTask(md.EbsCsiDriver); runningCsiTask != nil {
+		log.Debugf("engine ebs CSI driver is running with taskID: %v", runningCsiTask.GetID())
+	} else {
+		if ebsCsiDaemonManager, ok := w.taskEngine.GetDaemonManagers()[md.EbsCsiDriver]; ok {
+			if csiTask, err := ebsCsiDaemonManager.CreateDaemonTask(); err != nil {
+				// fail attachment and return
+				log.Errorf("Unable to start ebsCsiDaemon for task %s in the engine: error: %s", csiTask.GetID(), err)
+				csiTask.SetKnownStatus(apitaskstatus.TaskStopped)
+				csiTask.SetDesiredStatus(apitaskstatus.TaskStopped)
+				w.taskEngine.EmitTaskEvent(csiTask, err.Error())
+				return err
+			} else {
+				w.taskEngine.SetDaemonTask(md.EbsCsiDriver, csiTask)
+				w.taskEngine.AddTask(csiTask)
+				log.Infof("task_engine: Added EBS CSI task to engine")
+			}
+		} else {
+			log.Errorf("CSI Driver is not Initialized")
```

Comment:
Do we need to tell the backend that we failed to handle the EBS volume attachment?

Reply:
Or this shouldn't ever happen in practice and we just need to have some code to error the handle? I think the former?

Reply:
the backend has a timeout handler.  We can short-circuit that with an attachment update, but that's not part of the immediate plans.